design/trigger-backups-api-events.md

```diff
@@ -0,0 +1,35 @@
+# Trigger backups based on Kubernetes events 
+
+Triggering backups with Velero based on Kubernetes api events (Terminating, deleting, CrashLoopBackOff, etc..).
+
+## Goals
+
+- Rather than performing backups manually or by scheduling them, backups are created based on events.
+- Recovering the components of the cluster when they were deleted accidently. 
+- Blocking the delete process of a component in the cluster if it was a human intervention and trigger a backup.
+
+## Non Goals
+
+- N/A
+
+## Background
+
+## High-Level Design
+
+A custom controller that list and watch all the events of Kubenretes and based on that it trigger backups.
+
+## Detailed Design
+
+A custom controller where it have its own refelctor to list and watch a specific components using the kubernetes watch api. 
+It will add the object mentored with its status/current event to a queue.
+Then pop up the object based on its status (criticity and priority).
+Block the delete process if it was performed from a human being and trigger a backup for that component.
+Trying to catch the errors happening on the components so it predicts when they will be down and trigger backups.
```

Comment:
I'm not sure how this part connects to backups - there's something I'm missing.

Reply:
For example, we could use the help of other monitoring tools (promtail and loki), based on metrics we can for example predict if the application is going to be down (consuming many resources e.g memory, disk, etc..).

Reply:
I think that's getting outside of Velero's scope and into machine learning. While I think it's useful functionality for sure, I don't think I would accept this portion into Velero core.